libcurv/string.cc

```diff
@@ -25,6 +25,35 @@ maybe_string(Value val, const Context& cx)
 
 const char String::name[] = "string";
 
+#ifdef _WIN32
+    std::string
+    wstring_to_string(const std::wstring &wstr)
+    {
+        // Copied and modified from original Source: <https://stackoverflow.com/a/3999597>
+        // Author: tfinniga <https://stackoverflow.com/users/9042/tfinniga>
+        // License: CC BY-SA 3.0 <https://creativecommons.org/licenses/by-sa/3.0/>
+
+        if (wstr.empty())
+            return std::string();
+
+        // query wstring length first (!= byte count in general)
+        int size_needed = WideCharToMultiByte(CP_UTF8, 0, &wstr[0], (int) wstr.size(), NULL, 0, NULL, NULL);
+        if (size_needed == 0)
+        {
+            // given that we check for emptyness above, a return value of 0 must indicate an error here
+            // TODO: format GetLastError() using win_strerror (currently located in output_file.cc and unexposed) and throw exception
+        }
+
+        std::string strTo(size_needed, 0);
+        // now convert
+        if (WideCharToMultiByte(CP_UTF8, 0, &wstr[0], (int) wstr.size(), &strTo[0], size_needed, NULL, NULL) == 0) {
+            GetLastError();
+            // TODO: format GetLastError() using win_strerror (currently located in output_file.cc and unexposed) and throw exception
+        }
+        return strTo;
+    }
```

Comment:
@doug-moen Again, is it appropriate to put this function here? It's only used in string.h so far.